test/e2e/drain_tool_test.go

```diff
@@ -0,0 +1,283 @@
+	})
+
+	It("should place resources on all available member clusters", checkIfPlacedWorkResourcesOnAllMemberClusters)
+
+	It("create cluster resource placement disruption budget to block draining on all but one cluster", func() {
+		crpdb := placementv1beta1.ClusterResourcePlacementDisruptionBudget{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: crpName,
+			},
+			Spec: placementv1beta1.PlacementDisruptionBudgetSpec{
+				MinAvailable: &intstr.IntOrString{
+					Type:   intstr.Int,
+					IntVal: int32(len(allMemberClusterNames)) - 1,
+				},
+			},
+		}
+		Expect(hubClient.Create(ctx, &crpdb)).To(Succeed(), "Failed to create CRP Disruption Budget %s", crpName)
+	})
+
+	It("drain cluster using binary, should succeed", func() { runDrainClusterBinary(hubClusterName, memberCluster1EastProdName, true) })
+
+	It("drain cluster using binary, should fail due to CRPDB", func() { runDrainClusterBinary(hubClusterName, memberCluster2EastCanaryName, false) })
+
+	It("should ensure no resources exist on drained clusters", func() {
+		for _, cluster := range drainClusters {
+			resourceRemovedActual := workNamespaceRemovedFromClusterActual(cluster)
+			Eventually(resourceRemovedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to check if resources doesn't exist on member cluster")
+		}
+	})
+
+	It("should update cluster resource placement status as expected", func() {
+		crpStatusUpdatedActual := crpStatusUpdatedActual(workResourceIdentifiers(), noDrainClusterNames, nil, "0")
+		Eventually(crpStatusUpdatedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update cluster resource placement status as expected")
+	})
+
+	It("should still place resources on the selected clusters which were not drained", func() {
+		for _, cluster := range noDrainClusters {
+			resourcePlacedActual := workNamespaceAndConfigMapPlacedOnClusterActual(cluster)
+			Eventually(resourcePlacedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to place resources on the selected clusters")
+		}
+	})
+
+	It("uncordon cluster using binary", func() { runUncordonClusterBinary(hubClusterName, memberCluster1EastProdName) })
+
+	It("uncordon cluster using binary", func() { runUncordonClusterBinary(hubClusterName, memberCluster2EastCanaryName) })
+
+	It("should place resources on all available member clusters", checkIfPlacedWorkResourcesOnAllMemberClusters)
+})
+
+func runDrainClusterBinary(hubClusterName, memberClusterName string, isSuccess bool) {
+	By(fmt.Sprintf("draining cluster %s", memberClusterName))
+	cmd := exec.Command(drainBinaryPath,
+		"--hubClusterContext", hubClusterName,
+		"--clusterName", memberClusterName)
+	output, err := cmd.CombinedOutput()
+	Expect(err).ToNot(HaveOccurred(), "Drain command failed with error: %v\nOutput: %s", err, string(output))
+	if isSuccess {
```

Comment:
Same issue here; I am wondering if there is a better way to validate this rather than relying on log string matches.